libvast/vast/concept/parseable/vast/identifier.hpp

```diff
@@ -21,6 +21,6 @@
 
 namespace vast::parsers {
 
-constexpr inline auto identifier = +(alnum | ch<'_'>);
+constexpr inline auto identifier = +(alnum | ch<'_'> | ch<'.'>);
```

Comment:
If we all field names that include a `.`, flattening and unflattening will be broken unless we add extra syntactic elements. IIRC we discussed wrapping field names in single quotes.

Reply:
Well, changing the allowed input representation of the identifier parser wouldn't change its output. As soon as a field can have a dot in its name the unflattening heuristic will produce a nesting that wasn't present in the original input. The only real solution to this is to preserve the original layout structure in the database.